cpp/log/cert.cc

```diff
@@ -233,8 +262,8 @@ string Cert::PrintTime(ASN1_TIME* when) {
 }
 
 
-Cert::Status Cert::IsIdenticalTo(const Cert& other) const {
-  return X509_cmp(x509_, other.x509_) == 0 ? TRUE : FALSE;
+StatusOr<bool> Cert::IsIdenticalTo(const Cert& other) const {
+  return X509_cmp(x509_, other.x509_) == 0 ? true : false;
```

Comment:
Yes, that one doesn't return any errors. Might be better if I change it
just to make that clear.

Martin

On 28 August 2015 at 14:58, Pierre Phaneuf <notifications@github.com> wrote:

> In cpp/log/cert.cc
> <https://github.com/google/certificate-transparency/pull/885#discussion_r38201838>
> :
>
> > @@ -233,8 +262,8 @@ string Cert::PrintTime(ASN1_TIME* when) {
> >  }
> >
> >
> > -Cert::Status Cert::IsIdenticalTo(const Cert& other) const {
> > -  return X509_cmp(x509_, other.x509_) == 0 ? TRUE : FALSE;
> > +StatusOr<bool> Cert::IsIdenticalTo(const Cert& other) const {
> > +  return X509_cmp(x509_, other.x509_) == 0 ? true : false;
>
> Also, this could be a proper bool instead of a StatusOr<bool>, could make
> other call sites simpler?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/certificate-transparency/pull/885/files#r38201838>
> .
>